File: findMax.cpp
#include <vector>
#include <cstdio>
#include <ctime>
#pragma GCC optimize("Ofast")
#pragma GCC target("avx,avx2,fma")

int findMax(std::vector<int> a, std::vector<int> b) 
{
	printf("%d %d\n", int(a.size()), int(b.size()));
	int maxi = 0;
	for (int i = 0; i < int(a.size()); ++i) 
	{
		for (int j = 0; j < int(b.size()); ++j) 
		{
			if (i * j > maxi)
				maxi = i * j;
		}
	}
	return maxi;
}

int main()
{
	std::vector<int> a, b;
	for (int i = 0; i < 20000; ++i) 
	{
		a.emplace_back(0.5);
		b.emplace_back(0.4);
	}
	clock_t start, end;
	start = clock();
	findMax(a, b);
	end = clock();
	double time_taken = double(end - start) / double(CLOCKS_PER_SEC);
	printf("Time taken: %lf seconds\n", time_taken);
	return 0;
}

File: find_max.rs
use std::time::{Instant};

fn main() {
    let mut a: Vec<f32> = Vec::new();
    let mut b: Vec<f32> = Vec::new();
    for _i in 0..20000 {
        a.push(0.5);
        b.push(0.4);
    }
    let start = Instant::now();
    find_max(a, b);
    let duration = start.elapsed();
    println!("Time taken: {:?}", duration);
}

fn find_max(a: Vec<f32>, b: Vec<f32>)-> usize {
    println!("{} {}", a.len(), b.len());
    let mut maxi = 0;
    for i in 0..a.len() {
        for j in 0..b.len() {
            if i * j > maxi {
                maxi = i * j;
            }
        }
    }
    maxi
}
